src/main/kotlin/org/jetbrains/intellij/platform/gradle/artifacts/transform/ExtractorTransformer.kt

```diff
@@ -105,17 +115,76 @@ abstract class ExtractorTransformer @Inject constructor(
                     execOperations.exec {
                         commandLine("hdiutil", "detach", "-force", "-quiet", tempDirectory)
                     }
+
+                    createIvyXmls(tempDirectory)
                 }
 
-                else -> {}
+                else -> {
+                    createIvyXmls(targetDirectory)
+                }
             }
 
-            log.info("Extracting to '$targetDirectory' completed.")
+            log.warn("Extracting to '$targetDirectory' completed.")
         }.onFailure {
             log.error("${javaClass.canonicalName} execution failed.", it)
         }
     }
 
+    /**
+     * Pre-create Ivy XML files so that later this directory can be used as an Ivy repository without having to discover
+     * modules & plugins each time the build is run.
+     */
+    private fun createIvyXmls(platformPath: Path) {
+        val isIde = platformPath.listDirectoryEntries().map { it.name }.containsAll(
+            listOf("bin", "lib", "plugins", "product-info.json")
+        )
+        if (!isIde) {
+            log.info("The directory '$platformPath' is not an IDE, Ivy repository is not needed there.")
+            return
+        }
+
+        log.info("Creating an Ivy repository in '$platformPath'.")
+
+        val writtenIvyModules = HashSet<String>()
+        val productInfo = platformPath.productInfo()
+        val version = productInfo.buildNumber
+
+        val pluginManager = IdePluginManager.createManager(createTempDirectory())
+        val plugins = platformPath.getBundledPlugins(pluginManager)
+
+        for (plugin in plugins.values) {
```

Comment:
I think some optimizations are possible here for caching & skipping what was already discovered, but I do not want to over complicate it. Also since it is done in the transformer now, it does not matter that much.